digest-macro-derive/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+extern crate proc_macro;
+
+use crate::proc_macro::TokenStream;
+use quote::quote;
+use syn::{Data, Fields};
+
+#[proc_macro_derive(DigestRootMacro)]
+pub fn digest_root_macro_fn(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).unwrap();
+    impl_digest_root_macro(&ast)
+}
+
+fn impl_digest_root_macro(ast: &syn::DeriveInput) -> TokenStream {
+    let name = &ast.ident;
+    if let Data::Struct(data) = &ast.data {
+        if let Fields::Named(fields) = &data.fields {
+            let idents = fields
+                .named
+                .iter()
+                .map(|field| field.ident.as_ref().expect("Named field"));
+
+            let types = fields.named.iter().map(|field| &field.ty);
+
+            let gen = quote! {
+            impl DigestRoot for #name
```

Comment:
This is using types that it does not import, perhaps we need `use DigestRoot` here otherwise there is an added burden on the macro user to add these use statements?